src/net/rpc.rs

```diff
@@ -1512,37 +1511,66 @@ impl ConversationHttp {
     /// Load up the canonical Stacks chain tip.  Note that this is subject to both burn chain block
     /// Stacks block availability -- different nodes with different partial replicas of the Stacks chain state
     /// will return different values here.
-    /// tip_opt is given by the HTTP request as the optional query parameter for the chain tip
-    /// hash.  It will be None if there was no paramter given.
+    ///
+    /// # Inputs
+    /// - `tip_opt` is given by the HTTP request as the optional query parameter for the chain tip
+    /// hash.  It will be None if there was no parameter given.
+    /// - `use_latest_tip` is also an optional query parameter, and defaults to false.
+    ///
     /// The order of chain tips this method prefers is as follows:
-    /// * tip_opt, if it's Some(..),
-    /// * the unconfirmed canonical stacks chain tip, if initialized
-    /// * the confirmed canonical stacks chain tip
+    /// * If `use_latest_tip` is true, the method attempts to return the unconfirmed chain tip,
+    ///     and returns the confirmed stacks chain tip if that doesn't work.
+    /// * If `tip_opt` is Some(tip), the method returns tip.
+    /// * Otherwise, the function considers the canonical Stacks chain tip.
     fn handle_load_stacks_chain_tip<W: Write>(
         http: &mut StacksHttp,
         fd: &mut W,
         req: &HttpRequestType,
         tip_opt: Option<&StacksBlockId>,
         sortdb: &SortitionDB,
         chainstate: &StacksChainState,
+        use_latest_tip: bool,
     ) -> Result<Option<StacksBlockId>, net_error> {
-        match tip_opt {
-            Some(tip) => Ok(Some(*tip).clone()),
-            None => match chainstate.get_stacks_chain_tip(sortdb)? {
-                Some(tip) => Ok(Some(StacksBlockHeader::make_index_block_hash(
+        if use_latest_tip {
+            let unconfirmed_chain_tip_opt = match &chainstate.unconfirmed_state {
```

Comment:
I just want to confirm that the following scenarios are tested:

* The caller passes `use_latest_tip=1` with an existing unconfirmed microblock stream, and Clarity state from the unconfirmed microblock stream is successfully loaded.
* The caller passes `use_latest_tip=1` with an empty unconfirmed microblock stream, and Clarity state from the canonical chain tip is successfully loaded (i.e. you don't get a 404 since the unconfirmed chain tip points to a nonexistent MARF trie).
* The caller passes `use_latest_tip=0`, and the canonical chain tip is used regardless of the state of the unconfirmed microblock stream.

Reply:
That second bullet is going to be a bit thorny.  It may work in these unit tests, but the unit test framework doesn't reflect how the Stacks node actually deals with unconfirmed chain state.  This is because the Stacks node itself has a very weird way of representing unconfirmed state, due to the fact that unconfirmed state is curated both by the p2p thread (to service read-only queries off of it) and by the relayer thread (to process new microblocks).  I think you might want to create a nominal integration test for these bullet points in `neon_integrations.rs`.  It doesn't have to exhaustively test all of the RPC methods; it just needs to verify that the right chain tip is used in each situation (and nothing 404's).

Reply:
Addressed in commit 0c5f6aa